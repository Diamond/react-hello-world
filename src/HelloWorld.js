import React, { Component } from 'react';
import './HelloWorld.css';

class HelloWorld extends Component {
  constructor(props) {
    super(props);
    this.state = { greeting: 'Hello' };
    this.frenchify = this.frenchify.bind(this);
    this.removeGreeting = this.removeGreeting.bind(this);
  }
  frenchify() {
    this.setState({ greeting: 'Bonjour' });
  }
  removeGreeting() {
    this.props.removeGreeting(this.props.name);
  }
  render() {
    return (
      <div className="HelloWorld">
        {this.state.greeting} {this.props.name}!
        <br />
        <button onClick={this.frenchify}>Frenchify!</button>
        <br />
        <button onClick={this.removeGreeting}>Remove Me!</button>
      </div>
    );
  }
}

export default HelloWorld;
