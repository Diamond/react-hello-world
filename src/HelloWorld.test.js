import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import HelloWorld from './HelloWorld';

describe(HelloWorld, () => {
  const name = 'Person';
  const mockRemoveGreeting = jest.fn();
  const component = shallow(
    <HelloWorld name={name} removeGreeting={mockRemoveGreeting}/>
  );

  it('renders and matches our snapshot', () => {
    const component = renderer.create(
      <HelloWorld name="Person" />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('contains the supplied name', () => {
    expect(component.text()).toContain(name);
  });

  it('modifies the greeting when frenchify button is clicked', () => {
    component.find('button.frenchify').simulate('click');
    expect(component.text()).toContain('Bonjour');
  });

  it('calls the passed in removeGreeting function when remove button is clicked', () => {
    component.find('button.remove').simulate('click');
    expect(mockRemoveGreeting).toBeCalled();
  });
});
