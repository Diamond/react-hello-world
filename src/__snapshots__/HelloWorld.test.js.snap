exports[`class HelloWorld extends /* istanbul ignore next */_react.Component {
  constructor(props) {
    super(props);
    this.state = { greeting: 'Hello' };
    this.frenchify = this.frenchify.bind(this);
    this.removeGreeting = this.removeGreeting.bind(this);
  }
  frenchify() {
    this.setState({ greeting: 'Bonjour' });
  }
  removeGreeting() {
    this.props.removeGreeting(this.props.name);
  }
  render() {
    return (
      /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'div', /* istanbul ignore next */{ className: 'HelloWorld', __source: { fileName: _jsxFileName, lineNumber: 19 }, __self: this },
        this.state.greeting, /* istanbul ignore next */' ', this.props.name, /* istanbul ignore next */'!',
        /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'br', /* istanbul ignore next */{ __source: { fileName: _jsxFileName, lineNumber: 21 }, __self: this }),
        /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'button', /* istanbul ignore next */{ className: 'frenchify', onClick: this.frenchify, __source: { fileName: _jsxFileName, lineNumber: 22 }, __self: this }, /* istanbul ignore next */'Frenchify!'),
        /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'br', /* istanbul ignore next */{ __source: { fileName: _jsxFileName, lineNumber: 23 }, __self: this }),
        /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'button', /* istanbul ignore next */{ className: 'remove', onClick: this.removeGreeting, __source: { fileName: _jsxFileName, lineNumber: 24 }, __self: this }, /* istanbul ignore next */'Remove Me!')));


  }} renders and matches our snapshot 1`] = `
<div
  className="HelloWorld">
  Hello
   
  Person
  !
  <br />
  <button
    className="frenchify"
    onClick={[Function]}>
    Frenchify!
  </button>
  <br />
  <button
    className="remove"
    onClick={[Function]}>
    Remove Me!
  </button>
</div>
`;
